sgx/ias-types/src/nonce.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2018-2020 MobileCoin Inc.
+
+//! Nonce structures
```

Comment:
This is sliced from attest as well, the main difference is the use of repr bytes, and the implementation of `keys::FromRandom` instead of the more specific `Nonce` trait, which we used in attest.